build_tools/integration_tests/cpu_llm_server_test.py

```diff
@@ -0,0 +1,223 @@
+gpu_settings = {
+    "device_flags": [
+        "-iree-hal-target-backends=rocm",
+        f"--iree-hip-target={IREE_HIP_TARGET}",
+    ],
+    "device": "hip",
+}
+
+
+@pytest.fixture(scope="module")
+def model_test_dir(request, tmp_path_factory):
+    repo_id = request.param["repo_id"]
+    model_file = request.param["model_file"]
+    tokenizer_id = request.param["tokenizer_id"]
+    settings = request.param["settings"]
+    batch_sizes = request.param["batch_sizes"]
+
+    tmp_dir = tmp_path_factory.mktemp("cpu_llm_server_test")
+    hf_home = os.environ.get("HF_HOME", None)
+    hf_home = Path(hf_home) if hf_home is not None else tmp_dir
+    try:
+        # Download model if it doesn't exist
+        model_path = hf_home / model_file
+        if not os.path.exists(model_path):
+            subprocess.run(
+                f"huggingface-cli download --local-dir {hf_home} {repo_id} {model_file}",
+                shell=True,
+                check=True,
+            )
+
+        # Set up tokenizer if it doesn't exist
+        tokenizer_path = hf_home / "tokenizer.json"
+        if not os.path.exists(tokenizer_path):
+            tokenizer = AutoTokenizer.from_pretrained(
+                tokenizer_id,
+            )
+            tokenizer.save_pretrained(hf_home)
+
+        # Export model if it doesn't exist
+        mlir_path = tmp_dir / "model.mlir"
+        config_path = tmp_dir / "config.json"
+        bs_string = ",".join(map(str, batch_sizes))
+        subprocess.run(
+            [
+                "python",
+                "-m",
+                "sharktank.examples.export_paged_llm_v1",
+                f"--gguf-file={model_path}",
+                f"--output-mlir={mlir_path}",
+                f"--output-config={config_path}",
+                f"--bs={bs_string}",
+            ],
+            check=True,
+        )
+
+        # Compile model if it doesn't exist
+        vmfb_path = tmp_dir / "model.vmfb"
+        subprocess.run(
+            [
+                "iree-compile",
+                mlir_path,
+                "-o",
+                vmfb_path,
+            ]
+            + settings["device_flags"],
+            check=True,
+        )
+
+        # Write config if it doesn't exist
+        edited_config_path = tmp_dir / "edited_config.json"
+        config = {
+            "module_name": "module",
+            "module_abi_version": 1,
+            "max_seq_len": 2048,
+            "attn_head_count": 32,
+            "attn_head_dim": 100,
+            "prefill_batch_sizes": batch_sizes,
+            "decode_batch_sizes": batch_sizes,
+            "transformer_block_count": 26,
+            "paged_kv_cache": {"block_seq_stride": 16, "device_block_count": 256},
+        }
+        with open(edited_config_path, "w") as f:
+            json.dump(config, f)
+        yield hf_home, tmp_dir
+    finally:
+        shutil.rmtree(tmp_dir)
+
+
+@pytest.fixture(scope="module")
+def available_port(port=8000, max_port=8100):
```

Comment:
These fixtures / helper functions could move into a `conftest.py` or `test_utils.py` to share with other integration tests. Fine to keep here for now.

Reply:
Good idea, moved setup functions to `conftest.py`